server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -560,6 +561,33 @@ public void removeConnectionListener(TransportConnectionListener listener) {
                                                                 final TransportRequestOptions options,
                                                                 TransportResponseHandler<T> handler) {
         try {
+            if (request.getParentTask().isSet()) {
+                final Releasable unregisterChildNode = taskManager.registerChildNode(request.getParentTask().getId(), connection.getNode());
```

Comment:
I think that we need to actually capture the connection (not the DiscoveryNode object). The reason for this is cross-cluster connections. In the task cancellation action, we assume that the DiscoveryNode is sufficient to resolve the target of the cancellation. This only works for nodes in the same cluster, however.

I don't think that we need to solve this all in this PR, but we need to capture the issue properly, and add an item to the planning.

Reply:
++ see https://github.com/elastic/elasticsearch/pull/54312/commits/4f337decce08edd355621c9982559df34bc42ce6